link-grammar/sat-solver/word-tag.hpp

```diff
@@ -139,10 +139,9 @@ class WordTag
 
     if (len > UNLIMITED_LEN) len = UNLIMITED_LEN;
 
-    if (!_opts->all_short &&
-        (conset == NULL || match_in_connector_set(conset, c))) {
-      c->length_limit = UNLIMITED_LEN;
-    } else {
+    if (_opts->all_short ||
+        !(conset == NULL || match_in_connector_set(conset, c)))
```

Comment:
can you change this ti `(conset != NULL && !match_in_connector_set(conset, c))`  that would make it slightly easier to read ...